geth/peers/topicpool.go

```diff
@@ -0,0 +1,243 @@
+package peers
+
+import (
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common/mclock"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/p2p"
+	"github.com/ethereum/go-ethereum/p2p/discover"
+	"github.com/ethereum/go-ethereum/p2p/discv5"
+	"github.com/status-im/status-go/geth/params"
+)
+
+// NewTopicPool returns instance of TopicPool
+func NewTopicPool(topic discv5.Topic, limits params.Limits, slowSync, fastSync time.Duration) *TopicPool {
+	return &TopicPool{
+		topic:    topic,
+		limits:   limits,
+		slowSync: slowSync,
+		fastSync: fastSync,
+		peers:    map[discv5.NodeID]*peerInfo{},
+	}
+}
+
+// TopicPool manages peers for topic.
+type TopicPool struct {
+	topic    discv5.Topic
+	limits   params.Limits
+	slowSync time.Duration
+	fastSync time.Duration
+
+	quit    chan struct{}
+	running int32
+
+	mu         sync.RWMutex
+	discWG     sync.WaitGroup
+	consumerWG sync.WaitGroup
+	connected  int
```

Comment:
is that `isConnected` or number of connections?

Reply:
number